test/extensions/matching/input_matchers/cel_matcher/cel_matcher_test.h

```diff
@@ -0,0 +1,630 @@
+        }
+      }
+      args {
+        id: 4
+        const_expr {
+          string_value: "/foo"
+        }
+      }
+    }
+  }
+)pb";
+
+// Compiled CEL expression string: response.headers['content-type'] == 'text/plain'
+inline constexpr char ReponseHeadeCelExprString[] = R"pb(
+  expr {
+      id: 8
+      call_expr {
+        function: "_==_"
+        args {
+          id: 6
+          call_expr {
+            function: "_[_]"
+            args {
+              id: 5
+              select_expr {
+                operand {
+                  id: 4
+                  ident_expr {name: "response"}
+                }
+                field: "headers"
+              }
+            }
+            args {
+              id: 7
+              const_expr {
+                string_value: "content-type"
+              }
+            }
+          }
+        }
+        args {
+          id: 9
+          const_expr { string_value: "text/plain" }
+        }
+      }
+    }
+)pb";
+
+// Compiled CEL expression string: request.path == '/foo' && request.headers['user'] ==
+// 'staging'.
+// Note, source_info is not required for evaluation process that happens in data plane, which is
+// confirmed by test cases above. However, it will be included as part of result from parsing and
+// checking process passed to data plane. Thus, the entire compiled CEL result is also tested.
+inline constexpr char RequestHeaderAndPathCelString[] = R"pb(
+    expr {
+      id: 11
+      call_expr {
+        function: "_&&_"
+        args {
+          id: 3
+          call_expr {
+            function: "_==_"
+            args {
+              id: 2
+              select_expr {
+                operand {
+                  id: 1
+                  ident_expr {
+                    name: "request"
+                  }
+                }
+                field: "path"
+              }
+            }
+            args {
+              id: 4
+              const_expr {
+                string_value: "/foo"
+              }
+            }
+          }
+        }
+        args {
+          id: 9
+          call_expr {
+            function: "_==_"
+            args {
+              id: 7
+              call_expr {
+                function: "_[_]"
+                args {
+                  id: 6
+                  select_expr {
+                    operand {
+                      id: 5
+                      ident_expr {
+                        name: "request"
+                      }
+                    }
+                    field: "headers"
+                  }
+                }
+                args {
+                  id: 8
+                  const_expr {
+                    string_value: "user"
+                  }
+                }
+              }
+            }
+            args {
+              id: 10
+              const_expr {
+                string_value: "staging"
+              }
+            }
+          }
+        }
+      }
+    }
+    source_info {
```

Comment:
You can remove source_info, it's not used.

Reply:
Alright, that's reasonable. The compiler doesn't seem to read this fields at all but may in the future.